### PR TITLE
Yeganeh: Task 481-484

### PR DESCRIPTION
**Pull request by twlin037**
BANKING77
WebNLG

### CODE REVIEW

**Comment by swarooprm** [comment]
@yeganehkordi, wondering if you can help review this PR.

**Review by yeganehkordi — COMMENTED** [review]
Thanks for your work! 
The definition of the tasks seemed inadequate and unclear to me. Please take a look at some of the tasks and README file in this repo: https://github.com/allenai/natural-instructions-expansion/tree/master/tasks and update the tasks.
Also, please add the link to the dataset and paper in the source field.

**Comment by twlin037** [comment]
Excuse me, I don't see the link in the source field in your given task, can you give me a example to know what link to dataset do you want?

**Comment by yeganehkordi** [comment]
Yeah, I meant to provide the link to the dataset and paper you used in your tasks. For example:
    `"Source": [
        "PasiNLU (paper: https://arxiv.org/pdf/2012.06154.pdf) (dataset:https://github.com/persiannlp/parsinlu)"
    ],`

**Comment by swarooprm** [comment]
@twlin037, can you address reviewer comments so that I can evaluate?

**Comment by twlin037** [comment]
I'll do it later before 11 pm, sorry but I'm just a little busy

**Comment by twlin037** [comment]
@swarooprm @yeganehkordi I think I have added the description and the link for dataset, please check it, thank you.

**Review by yeganehkordi — COMMENTED** [review]
Thanks for the changes! 
Some general comments:
- Categories and summaries in the README file are a bit different from the tasks. Consider updating them after updating the tasks. Also, remove .json from the name of task 483 in the README.
- Make sure your definitions and explanations are grammatically correct and use proper punctuation.
- An average human should be able to do the tasks by just reading the definition. You can improve the quality of definitions by asking someone to read the definition and answer some of the instances. Their mistakes show the points that you should mention in the definition.
- There are some latin letters in the text(e.g., \u00f6 ). Consider using proper encoding (ensure_ascii=False) to make the JSON file human-readable.

Task 481:
- Please mention the classes in the definition. There are lots of classes in this task, and we need to clarify the difference between them in the definition.
- All the positive examples belong to the "card_arrival" class; consider using different classes in the examples.

Task 482:
- Please elaborate more on the `type of the sentence` and different classes.

Task 483:
- It might be hard to understand this task without reading the examples. Please make the definition more concise.

Task 484:
- In the second example, there is a repeated sentence `The runway length at Aarhus Airport is 2777.0 meters., The runway length at Aarhus airport is 2777.0.`. Consider removing repeated sentences from the instances.
- I'm not sure that this is a classification task. If there are limited classes, then it is a classification task; otherwise, it's most likely to be a text generation task. 
- Based on the definition in this task, we have to write the main topic of the text, but in some instances, we have several outputs for a sentence. So it doesn't seem to be the main topic. Please explain more about the task in the definition.

**Comment by twlin037** [comment]
Okay, so I should update again the file right?

**Comment by yeganehkordi** [comment]
Yes, please update the files.

**Comment by swarooprm** [comment]
> Yes, please update the files.

@twlin037, can you update soon so that I can evaluate?

**Comment by twlin037** [comment]
@yeganehkordi I think I have updated the json, please check, thank you

**Comment by swarooprm** [comment]
> @yeganehkordi I think I have updated the json, please check, thank you

@twlin037, can you fix tests so that @yeganehkordi  can check?

**Comment by twlin037** [comment]
@swarooprm I think I've fixed the problem, thanks you

**Comment by swarooprm** [comment]
@yeganehkordi, check if you are happy or have any other feedback?

**Comment by yeganehkordi** [comment]
Sorry for the delay! 
- I think in task 481, we still need all of the categories and their explanation in the definition. I understand this might be hard. So, what if we merge categories to have balance, exchange rate, card issues classes without going into details?
- There are some Latin letters in the text(e.g., \u00f6 ). Consider using proper encoding (ensure_ascii=False) to make the JSON file human-readable.
- Consider adding a space after periods.
- Most of the explanations seem to be missing their closing punctuations. Consider adding proper punctuations.
- Please replace ` specific department ` with ` the specific department ` in the definition of task 481.
- Task 481: How many classes do we have, and what are they? If it's not specified, then it's a text generation task.
- Task 482: Replace the definition with: `In this task, you are given a set of descriptions, and your task is to count how many different ways have been used to express the same thing. For example, 'John is my father' or 'I'm John's son' are talking about the same thing. Your answer should be a number.`
- Task 482: Replace `This is positive since it's counting is correct` with `This is positive since its counting is correct.` .
- Task 484: Maybe change the definition to sth like: `In this task, you are given a description, and you have to write the keywords about the description. There might be several keywords, but we only need you to write one of them.`, I think you can make this example better and more concise.
- Task 484: There are some repeated sentences in the input and repeated outputs. Consider removing them from the instances.
- Task 484: Please correct the typos of: `meters`, `sea level`, `runway length`, and `Airport is`.
- Task 848: Replace `This is negative since the text generated contains too many irrelated text. Also, it generate sentence instead of single text.` with `This is negative since the generated output contains too many unrelated words. Also, it generates sentences instead of a single word.`. I think it's better to use words in the explanations instead of text.

File: tasks/README.md [hunk 1]
```diff
@@ -6,4 +6,8 @@ This directory contains the tasks that are part of this benchmark.
 Name | Summary | Category
 ---- | ----------- | --------
 `task1001_squad1.1_question_generation` | Generating guestions (based on SQuAD 1.1) | Question Generation  
-`task1002_squad1.1_answer_generation` | Generating answers to SQuAD 1.1 questions | Answer Generation
\ No newline at end of file
+`task1002_squad1.1_answer_generation` | Generating answers to SQuAD 1.1 questions | Answer Generation
+`task481_BANKING77_classification` | Check the category of the banking query | Classification
```
Comment (yeganehkordi) [inline tasks/README.md:10]: Task 481: `Givan a sentence, classify the sentence based on the needed banking operation.`
Task 482: I'm not sure about this task. Please update the summary like the other examples after modifying the definition.
Task 483: `Givan a set of descriptions, write how many different ways have been used to express the same thing.`
Task 484: `Given a text, write a keyword about the text.`

**Comment by twlin037** [comment]
I'll modify tomorrow, let me finish my assignment tonight, thanks you

**Comment by twlin037** [comment]
@yeganehkordi I've modified but there's some error I can't handle:
First, I use ensure_ascii=False but it didn't work, sometimes it might also has the error `'cp950' codec can't encode character '\xa0' in position 18: illegal multibyte sequence
For repeated sentences, it's in the original input and it doesn't have the regulations so it's hard to remove them.
For the typos of: meters, sea level, runway length, and Airport is., there's the same problem. I can modify them by hand, but it's hard to handle it in the code since I don't know which kind of typo would appear.
`

**Comment by swarooprm** [comment]
> @yeganehkordi I've modified but there's some error I can't handle: First, I use ensure_ascii=False but it didn't work, sometimes it might also has the error `'cp950' codec can't encode character '\xa0' in position 18: illegal multibyte sequence` For repeated sentences, it's in the original input and it doesn't have the regulations so it's hard to remove them. For the typos of: meters, sea level, runway length, and Airport is., there's the same problem. I can modify them by hand, but it's hard to handle it in the code since I don't know which kind of typo would appear.

@yeganehkordi, check if you are happy or have any other feedback?

**Comment by yeganehkordi** [comment]
Thanks! I guess that's fine to go on without encoding in this case. All the typos that I've mentioned were in the explanations and definition. So, you don't need to handle them in the code.
Some missing comments:
- `There are 77 categories shown as below:0 for activate_my_card, ...`: I think it might be confusing to choose "0 "or "activate_my_card" as an output. So I suggest changing it to `There are 77 categories: activate_my_card, ...`
- Consider adding a space after punctuations in the definition and explanations.
- Most of the explanations seem to be missing their closing punctuations. Consider adding a period at the end of the sentences.
- Task 482: How many classes do we have, and what are they? If it's not specified, then it's a text generation task. Otherwise, mention all of the classes in the definition.
- Task 482: Please replace `it's description` with `its description` and `it's category` with `its category` in the explanations.
- Task 483: Please replace the definition with: `In this task, you are given a set of descriptions, and your task is to count how many different ways have been used to express the same thing. For example, 'John is my father' or 'I'm John's son' are talking about the same thing. Your answer should be a number.`
- Task 484: Maybe change the definition to sth like: `In this task, you are given a description, and you have to write the keywords about the description. There might be several keywords, but we only need you to write one of them.`, I think you can make this example better and more concise.

**Comment by yeganehkordi** [comment]
Hi @twlin037, Thanks for the updates.
Based on the [WebNLG Webpage](https://webnlg-challenge.loria.fr/challenge_2020/), we have some specific categories for task 482.

> The English WebNLG 2020 dataset for training comprises data-text pairs for 16 distinct DBpedia categories:
> The 10 seen categories used in 2017: Airport, Astronaut, Building, City, ComicsCharacter, Food, Monument, SportsTeam, University, and WrittenWork.
> The 5 unseen categories of 2017, which are now part of the seen data: Athlete, Artist, CelestialBody, MeanOfTransportation, Politician.

I guess you have used the 2017 version, which has 10 seen categories. So please add these categories and their explanation in the definition.
The rest of the tasks look good to me!

**Comment by twlin037** [comment]
@yeganehkordi I've made that change, please check.

**Comment by yeganehkordi** [comment]
Thanks! LGTM.

**Comment by swarooprm** [comment]
Your HW2 is complete now @twlin037

**Pull request closed without merging**